File: daisy1-core/src/main/java/io/openenterprise/daisy/service/ActionService.java
package io.openenterprise.daisy.service;

import io.openenterprise.daisy.InvocationContext;
import io.openenterprise.daisy.domain.Action;
import jakarta.annotation.Nonnull;
import jakarta.annotation.Nullable;

import java.util.List;
import java.util.UUID;

public interface ActionService {

    @Nullable
    <T> T invoke(@Nonnull InvocationContext invocationContext, @Nonnull Action<T> action);

    @Nullable
    <T> T invoke(@Nonnull UUID invocationContextId, @Nonnull Action<T> action);

    @Nullable
    Object invoke(@Nonnull InvocationContext invocationContext, @Nonnull List<Action<?>> actions);

    @Nullable
    Object invoke(@Nonnull UUID invocationContextId, @Nonnull List<Action<?>> actions);
}

File: daisy1-data/src/main/kotlin/io/openenterprise/daisy/data/jpa/service/ActionService.kt
package io.openenterprise.daisy.data.jpa.service

import io.openenterprise.daisy.data.jpa.domain.Action
import io.openenterprise.daisy.data.service.AuditableService
import jakarta.inject.Named
import java.time.Instant
import java.util.UUID

@Named
interface ActionService: AuditableService<Action, UUID, String, Instant>
